xdsl/dialects/transform.py

```diff
@@ -1,41 +1,325 @@
+    """
+
     name = "transform.any_param"
 
 
 @irdl_attr_definition
-class AnyValueType(ParametrizedAttribute, TypeAttribute):
+class AnyValueType(TransformHandleType):
+    """
+    https://mlir.llvm.org/docs/Dialects/Transform/#anyvaluetype
+    """
+
     name = "transform.any_value"
 
 
+@irdl_attr_definition
+class OperationType(TransformHandleType):
+    """
+    https://mlir.llvm.org/docs/Dialects/Transform/#operationtype
+    """
+
+    name = "transform.op"
+    operation: ParameterDef[StringAttr]
+
+    def __init__(self, operation: str):
+        super().__init__(parameters=[StringAttr(operation)])
+
+
+@irdl_attr_definition
+class ParamType(TransformHandleType):
+    """
+    https://mlir.llvm.org/docs/Dialects/Transform/#paramtype
+    """
+
+    name = "transform.param"
+    type: ParameterDef[TypeAttribute]
+
+    def __init__(self, type: TypeAttribute):
+        super().__init__(parameters=[type])
+
+
+@irdl_attr_definition
+class TypeParamType(TransformHandleType):
+    """
+    https://mlir.llvm.org/docs/Dialects/Transform/#typeparamtype
+    """
+
+    name = "transform.type"
+
+
+class FailurePropagationModeType(StrEnum):
+    PROPAGATE = "propagate"
+    SUPPRESS = "suppress"
+
+
+@irdl_attr_definition
+class FailurePropagationModeAttr(EnumAttribute[FailurePropagationModeType]):
+    name = "transform.failures"
+
+
+@irdl_op_definition
+class YieldOp(IRDLOperation):
+    """
+    https://mlir.llvm.org/docs/Dialects/Transform/#transformyield-transformyieldop
+    """
+
+    name = "transform.yield"
+
+    traits = frozenset([IsTerminator()])
+
+
+@irdl_op_definition
+class SequenceOp(IRDLOperation):
+    """
+    https://mlir.llvm.org/docs/Dialects/Transform/#transformsequence-transformsequenceop
+    """
+
+    name = "transform.sequence"
+
+    # TODO: Find out how to also use the enum FailurePropagationModeAttr as well as AnyIntegerAttr
+
+    body = region_def("single_block")
+    failure_propagation_mode = prop_def(AnyIntegerAttr)
+
+    root = var_operand_def(AnyOpType)
+    extra_bindings = var_operand_def(TransformHandleType)
+
+    irdl_options = [AttrSizedOperandSegments(as_property=True)]
+    traits = frozenset([IsolatedFromAbove()])
+
+    def __init__(
+        self,
+        failure_propagation_mode: FailurePropagationModeAttr | Attribute,
+        root: Sequence[SSAValue],
+        extra_bindings: Sequence[SSAValue],
+        body: Region,
+    ):
+        super().__init__(
+            properties={
+                "failure_propagation_mode": failure_propagation_mode,
+            },
+            regions=[body],
+            operands=[root, extra_bindings],
+        )
+
+
+@irdl_op_definition
+class TileOp(IRDLOperation):
+    """
+    https://mlir.llvm.org/docs/Dialects/Transform/#transformstructuredtile_using_for-transformtileusingforop
+    """
+
+    name = "transform.structured.tile"  # "transform.structured.tile_using_for" as of mlir 18.0
```

Comment:
I would say let's stick to the name as it is in the commit version that xDSL is coupled with, it's a little confusing to have both of these at the same time. When we update the MLIR commit, we will need to change this name to whichever is more appropriate.